meta-resin-common/recipes-containers/docker-disk/files/entry.sh

```diff
@@ -8,9 +8,10 @@ DOCKER_TIMEOUT=20 # Wait 20 seconds for docker to start
 # Default values
 PARTITION_SIZE=${PARTITION_SIZE:=1024}
 
-# Create a BTRFS disk image
-dd if=/dev/zero of=/export/data_disk.img bs=1M count=$PARTITION_SIZE
-mkfs.btrfs --mixed --metadata=single /export/data_disk.img
+# Create sparse file to hold ext4 resin-data partition
+dd if=/dev/zero of=/export/data_disk.img bs=1M count=0 seek=$PARTITION_SIZE
+# now partition the newly created file to ext4
+mkfs.ext4 /export/data_disk.img
```

Comment:
We should pass some parameters here. Specifically `-E lazy_itable_init=0` and `-E lazy_journal_init=0` so that a full initialisation of the filesystem happens at build time.